app/models/mixins/external_url_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ExternalUrlMixin
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :external_urls, :as => :refers_to, :dependent => :destroy
+  end
+
+  def external_url=(url, user)
+    external_urls.where(:user => user).destroy_all
+    external_url = ExternalUrl.create!(:refers_to => self, :url => url, :user => user)
```

Comment:
No need to set `external_url` variable, just return the result of the `create!` method.